requirements/customer/cli.tex

```diff
@@ -0,0 +1,102 @@
+\newcommand{\jacommand}[3]{
+
+\textsc{\large{\textbf{#1}}}
+\begin{itemize}
+\item[\textbf{Options:}] #2
+\item[\textbf{Effect:}] #3
+\end{itemize}
+}
+\newcommand{\jaoption}[3]{
+\begin{itemize}
+\item[] \textbf{#1} (#2)
+\item[\textbf{Effect:}] #3
+\end{itemize}
+}
+\subsection{Command Line Interface}
+All components of JobAdder are accessible through a command line interface (CLI).
+The command line interface on a machine with JobAdder installed has the following structure:
+\begin{equation}
+\text{<component name> <command> <options> <targets>},
+\end{equation}
+where the \textit{component name} can be any one of the following:
+\begin{itemize}
+\item \textbf{jobadder} to access the \textit{user client}.
+\item \textbf{jobcenter} to access the \textit{server}.
+\item \textbf{jobworker} to access the \textit{worker client}.
+\end{itemize}
+Commands, options, and valid targets are and options are specified in the following sections.
+
+Options can be added in the following ways:
+\begin{itemize}
+\item By supplying their names like "--<option\_x> --<option\_y>=2 --<option\_z>".
+\item By supplying their abbreviations like "-x -y 2 -z".
+\item By supplying their abbreviations like "-xz -y 2".
+\end{itemize}
+
+The available \textit{commands} are generally specific to the used component.
+Likewise, the available \textit{options} and \textit{targets} are generally specific to the commands of a used component.
+However, some commands and options are shared between components or commands.
+
+The JobAdder CLI ignores capitalization.
```

Comment:
Why? There are a lot of CLI programs where for ex. `-e` and `-E` are different options.

Reply:
I think that we should avoid capitalization if possible because it will make it harder to memorize the options available.
So far there was no need to use upper-case letters for options.
Therefore I think that as long as we do not need to add upper-case options we should just ignore capitalization alltogether.

Reply:
I do not like it nevertheless, as I said, ignoring capitalization isn't something standard on *nix

Reply:
I think you're right.
I'll still try to avoid upper-case letters but the CLI should not be case insensitive.